src/MusicStore/Components/GenreMenuComponent.cs

```diff
@@ -19,20 +19,9 @@ public GenreMenuComponent(MusicStoreContext dbContext)
 
         public async Task<IViewComponentResult> InvokeAsync()
         {
-            var genres = await GetGenres();
+            var genres = await DbContext.Genres.Select(g => g.Name).Take(9).ToListAsync();
```

Comment:
Just leaving a thought here for posterity (not actually suggesting a change).

If in the future we do want to order these by quantity (see TODO) then we could just wrap this in an #CacheTagHelper to prevent it from reexecuting on every page load.